tee-worker/app-libs/stf/src/trusted_call.rs

```diff
@@ -471,124 +462,125 @@ where
 				info!("Trying to create evm contract with address {:?}", contract_address);
 				Ok(())
 			},
-			// litentry
-			TrustedCall::set_user_shielding_key_preflight(root, who, key, hash) => {
-				if let Err(e) =
-					Self::set_user_shielding_key_preflight(root, shard, who.clone(), key, hash)
-				{
-					debug!("set_user_shielding_key_preflight error: {}", e);
-					add_call_from_imp_error(
-						calls,
-						node_metadata_repo,
-						Some(SgxParentchainTypeConverter::convert(who)),
-						e.to_imp_error(),
-						hash,
-					);
+			// litentry trusted calls
+			// the reason that most calls have an internal handling fn is that we want to capture the error and
+			// handle it here to be able to send error events to the parachain
+			TrustedCall::set_user_shielding_key(signer, who, key, hash) => {
+				debug!("set_user_shielding_key, who: {}", account_id_to_string(&who));
+				let parent_ss58_prefix =
+					node_metadata_repo.get_from_metadata(|m| m.system_ss58_prefix())??;
+				let account = SgxParentchainTypeConverter::convert(who.clone());
+				let c = node_metadata_repo
+					.get_from_metadata(|m| m.user_shielding_key_set_call_indexes())??;
+
+				match Self::set_user_shielding_key_internal(
+					signer,
+					who.clone(),
+					key,
+					parent_ss58_prefix,
+				) {
+					Ok(key) => {
+						debug!("pushing user_shielding_key_set event ...");
+						let id_graph =
+							IMT::get_id_graph_with_max_len(&who, RETURNED_IDGRAPH_MAX_LEN);
+						calls.push(OpaqueCall::from_tuple(&(
+							c,
+							account,
+							aes_encrypt_default(&key, &id_graph.encode()),
+							hash,
+						)));
+					},
+					Err(e) => {
+						debug!("pushing error event ... error: {}", e);
+						add_call_from_imp_error(
+							calls,
+							node_metadata_repo,
+							Some(account),
+							e.to_imp_error(),
+							hash,
+						);
+					},
```

Comment:
Looks pretty similar across cases; can we factor it out? :thinking:

Reply:
That's already one-liner (of a function call). If we want to make it even shorter, then macro is probably the only way. But normally macro would reduce the readability, so IMO it's a style issue.